administrator/components/com_media/resources/scripts/store/state.js

```diff
@@ -7,20 +7,32 @@ if (options.providers === undefined || options.providers.length === 0) {
 // The initial state
 export default {
     // Will hold the activated filesystem disks
-    disks: options.providers.map((disk) => {
-        return Object.assign(disk, {
-            root: disk.name + ':/',
-        })
+    disks: options.providers.map((provider) => {
+        let result = {};
+        result.displayName = provider.displayName;
+        result.accounts = [];
+
+        for(let i = 0; i < provider.adapterNames.length; i++)
+        {
+            let account = {
+                root : provider.name + '-' + i + ':/',
+                displayName : provider.adapterNames[i],
+            };
+
+            result.accounts.push(account);
+        }
+
+        return result;
     }),
     // The loaded directories
     directories: options.providers.map((disk) => {
-        return {path: disk.name + ':/', name: disk.displayName, directories: [], files: [], directory: null}
+        return {path: disk.name + '-0:/', name: disk.displayName, directories: [], files: [], directory: null}
```

Comment:
Yes, but we should not hardcode that in. This should be passed to the client from the server. Frontend does not need to know that much about how the server handles multiple accounts.